src/osemgrep/cli_scan/Scan_subcommand.mli

```diff
@@ -11,11 +11,20 @@ val main : string array -> Exit_code.t
 val run_conf : Scan_CLI.conf -> Exit_code.t
 val run_scan_conf : Scan_CLI.conf -> Exit_code.t
 
+type diff_config = { diff_targets : Fpath.t list; diff_depth : int option }
```

Comment:
This require more comments. What is diff_depth for example?
And why we need an extra type for those kinds of targets? We already have
functions that accept a list of target, so why we must care whether the targets
come from a diff or from another way?

Reply:
Done.